Formula/micromamba.rb

```diff
@@ -0,0 +1,83 @@
+class Micromamba < Formula
+  desc "Fast Cross-Platform Package Manager"
+  homepage "https://github.com/mamba-org/mamba"
+  url "https://github.com/mamba-org/mamba/archive/refs/tags/2023.02.03.tar.gz"
+  version "1.3.0"
+  sha256 "4c794246dd6008caebebef8c6266e6ba3f9ff5ce48908c830e448811587e9921"
+  license "BSD-3-Clause"
+
+  depends_on "cmake" => :build
+  depends_on "cli11"
+  depends_on "fmt"
+  depends_on "libsolv"
+  depends_on "libssh2"
+  depends_on "lz4"
+  depends_on "nlohmann-json"
+  depends_on "openssl@1.1"
+  depends_on "reproc"
+  depends_on "spdlog"
+  depends_on "tl-expected"
+  depends_on "xz"
+  depends_on "yaml-cpp"
+  depends_on "zstd"
+
+  uses_from_macos "curl", since: :monterey # uses CURLINFO_RETRY_AFTER, available since curl 7.66.0
+  uses_from_macos "krb5"
+  uses_from_macos "libarchive", since: :monterey
+  uses_from_macos "zlib"
+
+  resource "libarchive-headers" do
+    on_macos do
+      url "https://github.com/apple-oss-distributions/libarchive/archive/refs/tags/libarchive-113.tar.gz"
+      sha256 "b422c37cc5f9ec876d927768745423ac3aae2d2a85686bc627b97e22d686930f"
+    end
+  end
+
+  def install
+    if OS.mac?
+      resource("libarchive-headers").stage do
+        cd "libarchive/libarchive" do
+          (libexec/"include").install "archive.h", "archive_entry.h"
+        end
+      end
+    end
+
+    args = %w[
+      -DBUILD_LIBMAMBA=ON
+      -DBUILD_SHARED=ON
+      -DBUILD_MICROMAMBA=ON
+      -DMICROMAMBA_LINKAGE=DYNAMIC
+    ]
+
+    args << "-DCMAKE_INSTALL_RPATH=#{rpath}"
+
+    if OS.mac?
+      args << "-DLibArchive_INCLUDE_DIR=#{libexec}/include"
+      args << "-DCMAKE_CXX_FLAGS=-I#{libexec}/include"
```

Comment:
```suggestion
      ENV.append_to_cflags "-I#{libexec}/include"
```
Let's not use `CMAKE_CXX_FLAGS`, since this will make CMake ignore any `CXXFLAGS` environment variables.